monitoring/openshift/cronjob-registry-prune.yaml

```diff
@@ -0,0 +1,150 @@
+    description: "Name of the Scheduled Job to Create."
+    value: "cronjob-registry-soft-prune"
+    required: true
+  - name: "JOB_SOURCE_SCRIPT"
+    displayName: "Job Source Script"
+    description: "Name of the Scheduled Job's ConfigMap script to Create."
+    value: "soft-prune-script"
+    required: true
+  - name: "PRUNE_KEEP"
+    displayName: "Prune Keep Options"
+    description: "Keep options to pass to the prune job.  eg: --keep-younger-than=96h"
+    required: false
+    value: "--keep-younger-than=96h"
+  - name: "SCHEDULE"
+    displayName: "Cron Schedule"
+    description: "Cron Schedule to Execute the Job (in UTC)"
+    value: "0 5 * * *"
+#    value: "*/2 * * * *"
+    required: true
+  - name: "JOB_SERVICE_ACCOUNT"
+    displayName: "Service Account Name"
+    description: "Name of the Service Account To Exeucte the Job As."
+    value: "registry-pruner"
+    required: true
+  - name: "SUCCESS_JOBS_HISTORY_LIMIT"
+    displayName: "Successful Job History Limit"
+    description: "The number of successful jobs that will be retained"
+    value: "5"
+    required: true
+  - name: "FAILED_JOBS_HISTORY_LIMIT"
+    displayName: "Failed Job History Limit"
+    description: "The number of failed jobs that will be retained"
+    value: "5"
+    required: true
+  - name: "JOB_BACKOFF_LIMIT"
+    displayName: "Job Backoff Limit"
+    description: "The number of attempts to try for a successful job outcome (default: 6)"
+    value: "1"
+    required: false
+objects:
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+    name: ${JOB_SOURCE_SCRIPT}
+    namespace: ${NAMESPACE}
+    labels:
+      template: "cronjob-registry-prune"
+  data:
+    ${JOB_SOURCE_SCRIPT}: |
+      NOOP=echo
+      echo "running: oc adm prune images --keep-younger-than=96h"; date
+      oc adm prune images --keep-younger-than=96h
+      rc=$?; if [ $rc -ne 0 ] ; then
+        date
+        echo "sleeping... zzz"
+        sleep 60s
+        echo "Dry-Run Prune failed, aborting..."
+        exit $rc;
+      fi
+      ${NOOP} oc adm prune images --keep-younger-than=96h
+      rc=$?; if [ $rc -ne 0 ] ; then
+        date
+        echo "sleeping... zzz"
+        sleep 60s
+        echo "Prune failed, aborting..."
+        exit $rc;
+      fi
+      ${NOOP} oc rollout latest dc/docker-registry -n default
+      ${NOOP} oc rollout status dc/docker-registry -n default
+- kind: "CronJob"
+  apiVersion: "batch/v1beta1"
+  metadata:
+    name: "${JOB_NAME}"
+    namespace: ${NAMESPACE}
+    labels:
+      template: "cronjob-registry-prune"
+  spec:
+    schedule: "${SCHEDULE}"
+    concurrencyPolicy: "Forbid"
+    successfulJobsHistoryLimit: "${{SUCCESS_JOBS_HISTORY_LIMIT}}"
+    failedJobsHistoryLimit: "${{FAILED_JOBS_HISTORY_LIMIT}}"
+    jobTemplate:
+      spec:
+        backoffLimit: ${JOB_BACKOFF_LIMIT}
+        template:
+          spec:
+            containers:
+              - name: "${JOB_NAME}"
+                image: "openshift/jenkins-slave-base-centos7"
```

Comment:
`openshift3/ose:v3.10.34` seems to work as well. Not sure we need a pod with java? Think its worth locking this down to match the installed OCP version?